src/libre/re.c

```diff
@@ -103,11 +105,27 @@ re_comp(enum re_dialect dialect, int (*getc)(void *opaque), void *opaque,
 		return NULL;
 	}
 
-	new = m->comp(getc, opaque, opt, flags, m->overlap, err);
-	if (new == NULL) {
+	ast = m->parse(getc, opaque, opt, flags, m->overlap, err);
+	if (ast == NULL) {
 		return NULL;
 	}
 
+	/* Do a complete pass over the AST, filling in other details. */
```

Comment:
Now that parse has finished and the AST is complete, we can do a full top-down sweep over the entire AST for other details that difficult to decide locally, at parse time.

Currently, this pass just numbers the match group nodes, but soon it will also set info about ^ and $ anchoring (which depends on anchoring details for all alts in any `(x|y|z)` group), and could also do other rewriting.